main/src/ca/uwaterloo/flix/language/phase/Lowering.scala

```diff
@@ -1641,6 +1648,31 @@ object Lowering {
     LoweredAst.Expression.Cast(blockExp, None, Some(Type.Pure), Some(Type.Empty), tpe, pur, eff, loc.asSynthetic)
   }
 
+  /**
+    * Returns the expression of a `ParYield` expression that should be evaluated in the calling thread.
+    */
+  private def mkParYieldMainThread(exps: List[LoweredAst.ParYieldFragment], waitExps: LoweredAst.Expression): LoweredAst.Expression = {
+    exps.foldRight(waitExps) {
+      case (exp, acc) => mkLetMatch(exp.pat, exp.exp, acc)
+    }
+  }
+
+  /**
+    * Returns `true` if the ParYield fragment should be spawned in a thread. Wrapper for `isSimple`.
+    */
+  private def spawnable(frag: LoweredAst.ParYieldFragment): Boolean = !isSimple(frag.exp)
+
+  /**
+    * Returns `true` if `exp0` is either a literal or a variable.
+    */
+  @tailrec
+  private def isSimple(exp0: LoweredAst.Expression): Boolean = exp0 match {
+    case LoweredAst.Expression.Var(_, _, _) => true
+    case LoweredAst.Expression.Cst(_: Ast.Constant, _, _) => true
+    case LoweredAst.Expression.Scope(_, _, exp, _, _, _, _) => isSimple(exp)
```

Comment:
Lets not do this. For regions we leave it alone.

Reply:
Oh yes of course. For some reason I had this confused with a block exp i.e. `{ exp }`